### ISSUE TITLE
Install custom CA certificate?

### ISSUE THREAD

**Issue by larsks** [democratic-csi/democratic-csi#89]
Our TrueNAS server is using a certificate signed by a local certificate authority. I'd like to be able to use:

```
        httpConnection:
          allowInsecure: true
          apiKey: ...
          host: truenas.local
          port: 443
          protocol: https
          allowInsecure: false
```

But:

- I'm not sure how to tell democratic-csi about the appropriate CA certificate, and 
- I'm not sure if there's a way to map the hostname to an ip address if the hostname isn't in DNS (on my local system I would just edit `/etc/hosts`, but not sure what the appropriate solution is in this case).

Is there a graceful away of accomplishing both of these tasks?

**Comment by travisghansen** [comment]
Yeah there are..both need to be added to the chart though (help welcome :) )

- https://github.com/travisghansen/external-auth-server/blob/master/charts/external-auth-server/templates/deployment.yaml#L116
- https://kubernetes.io/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases/

The first is just an example of how to implement it in a chart...ultimately we just need to point that special env var to a file containing the certs.

**Comment by larsks** [comment]
Excellent, thank you!

Would the `hostAliases` need to be added to the node pods, the controller pod, or all of the above?

**Comment by travisghansen** [comment]
It should only be required in the `csi-driver` container of the `controller`.

**Comment by travisghansen** [comment]
I just published chart version `0.8.3` https://github.com/democratic-csi/charts/commit/34bb7d58a2fa8d8b7fc5a0963121cffd084f0b21 which includes support for `hostAliases` and `extraEnv`. Use that in conjunction with the `extraVolumes` and `extraVolumeMounts` and you should be good to go.
